sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/SimplifySizeExpressionCheck.kt

```diff
@@ -0,0 +1,234 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.kotlin.checks
+
+import org.jetbrains.kotlin.com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.lexer.KtToken
+import org.jetbrains.kotlin.lexer.KtTokens.ANDAND_Id
+import org.jetbrains.kotlin.lexer.KtTokens.EQEQ_Id
+import org.jetbrains.kotlin.lexer.KtTokens.EXCLEQ_Id
+import org.jetbrains.kotlin.lexer.KtTokens.EXCL_Id
+import org.jetbrains.kotlin.lexer.KtTokens.GT_Id
+import org.jetbrains.kotlin.lexer.KtTokens.LT_Id
+import org.jetbrains.kotlin.lexer.KtTokens.OROR_Id
+import org.jetbrains.kotlin.psi.KtBinaryExpression
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.KtConstantExpression
+import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtNameReferenceExpression
+import org.jetbrains.kotlin.psi.KtParenthesizedExpression
+import org.jetbrains.kotlin.psi.KtPrefixExpression
+import org.jetbrains.kotlin.psi.KtReferenceExpression
+import org.jetbrains.kotlin.psi.KtUnaryExpression
+import org.jetbrains.kotlin.psi.psiUtil.isNull
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.calls.model.ResolvedCall
+import org.sonar.check.Rule
+import org.sonarsource.kotlin.api.ArgumentMatcher
+import org.sonarsource.kotlin.api.CallAbstractCheck
+import org.sonarsource.kotlin.api.FunMatcher
+import org.sonarsource.kotlin.api.FunMatcherImpl
+import org.sonarsource.kotlin.api.fieldMatcher
+import org.sonarsource.kotlin.api.predictRuntimeIntValue
+import org.sonarsource.kotlin.plugin.KotlinFileContext
+
+private const val packageKotlinCollection = "kotlin.collections"
+private const val packageKotlinText = "kotlin.text"
+
+private const val interfaceKotlinCollection = "$packageKotlinCollection.Collection"
+private const val interfaceKotlinMap = "$packageKotlinCollection.Map"
+
+private val emptyArguments = listOf(emptyList<ArgumentMatcher>())
+
+private val countMatcher = FunMatcher(
+    definingSupertypes = setOf(packageKotlinCollection, packageKotlinText),
+    name = "count",
+    arguments = emptyArguments
+)
+
+private val isEmptyMatcher = FunMatcher(
+    definingSupertypes = setOf(interfaceKotlinCollection, interfaceKotlinMap, packageKotlinText),
+    name = "isEmpty",
+    arguments = emptyArguments
+)
+
+private val isNotEmptyMatcher = FunMatcher(
+    definingSupertypes = setOf(packageKotlinCollection, packageKotlinText),
+    name = "isNotEmpty",
+    arguments = emptyArguments
+)
+
+private val sizeFieldMatcher = fieldMatcher {
```

Comment:
I'm not sure I like this syntax, I'd rather make it 2 separate matchers,

Reply:
Fixed.